gst/nnstreamer/nnstreamer_plugin_api_util_impl.c

```diff
@@ -1131,10 +1152,8 @@ gst_tensor_dimension_string_is_equal (const gchar * dimstr1,
     if (!rank1 || !rank2)
       goto done;
 
-    for (j = 0; j < NNS_TENSOR_RANK_LIMIT; j++) {
-      if (dim1[j] != dim2[j])
-        goto done;
-    }
+    if (!gst_tensor_dimension_is_equal (dim1, dim2))
```

Comment:
```rank1``` and ```rank2``` are gathered from ```dim1``` and ```dim2```. How about checking the equivalent of them before getting ```rank1``` and ```rank2```?